### DISCUSSION TITLE
Blink-Camera stream to frigate

### DISCUSSION THREAD

**Discussion by Sharky84** [blakeblackshear/frigate#8935] [Ask A Question]
Hallo,

I have some Blink-cameras running. And I'm trying to connect them to the wonderful frigate nvr system.
My first attempts were to download periodically the mp4-files via the library blinkpy and stream them with simple-rtsp-server to frigate. But this has a big delay and sometimes frigate misses a video or gets it double from the rtsp-server.

Now, I turned a raspberry pi into a usb-mass-storage device and connected it with the Blink-Sync-module. The Blink-sync-module accepts it as an mass-storage device. So the mp4-files are available via the rpi-linux-operation system quite immediately.
What would be the best way to stream them to frigate, so frigate won't miss a video or gets it double?

Thank you very much in advance for your ideas and help

Michael

**Comment by banthungprong** [comment]
I stream my cams with libcamera and go2rtc to frigate. Should work with the file option of go2rtc too.

Use the arm executable of go2rtc. Define your stream with the file option in a yaml-file. 
Use a systemd service like this one (put exe and yaml in the directories from this file or change the directories):

> [Unit]
> Wants=network.target
> [Service]
> ExecStart=/usr/local/bin/go2rtc_linux_arm -config /usr/local/etc/go2rtc.yaml
> [Install]
> WantedBy=multi-user.target

**Reply by Sharky84** [reply]
Thank you for your quick answer. I will try this out.

**Reply by heapxor** [reply]
@Sharky84 any luck?
also how can u force blink cameras to constantly stream - not only in the case when motion is detected?

thanks!

**Reply by Sharky84** [reply]
@banthungprong
I didn't get it to work. I tried finally with mediamtx. The streaming is working, but frigate still misses some videos because it disconnects to the server when it gets no input (when there is not a stream - no motion):
rtsp://192.168.1.228:8554/haustuer: Server returned 400 Bad Request
 [2023-12-23 14:47:24] watchdog.Hoftor                ERROR   : Ffmpeg process crashed unexpectedly for Hoftor. 

Can I keep a "black" video streaming while there is no motion?

**Reply by Sharky84** [reply]
@heapxor 
Not really luck:-(
I think there is a url for constantly streaming and maybe you can keep the battery powered Blink cameras running with the USB-connected to a power supply.
But actually I don't know the url. Sorry.

**Comment by firefoot007** [comment]
There is a nice project at https://github.com/roger-/blinkbridge which can do exactly this. It streams the last frame of the video constantly so  Frigate ffmpeg does not disconnect.
